### PR TITLE
fix npe in query traverser when default variable value is a list

### PR DESCRIPTION
**Pull request by temaEmelyan**
inside of `visitValue` method we check whether a parent of a node is a var reference
```java
if (context.getParentNode() instanceof VariableDefinition) {
    return TraversalControl.CONTINUE;
}
```

But if a grandparent of a node is a var reference we won't exit the method. The method fails with NPE

### CODE REVIEW

File: src/test/groovy/graphql/analysis/QueryTraverserTest.groovy [hunk 1]
```diff
@@ -581,6 +581,62 @@ class QueryTraverserTest extends Specification {
...
+        given:
+        def schema = TestUtil.schema("""
+            type Query{
+                foo(arg1: [String]): String
+            }
+        """)
+        def visitor = mockQueryVisitor()
+        def query = createQuery("""
+            query myQuery(\$myVar: [String] = ["hello default"]) {foo(arg1: \$myVar)} 
+            """)
```
Comment (bbakerman) [inline src/test/groovy/graphql/analysis/QueryTraverserTest.groovy:595]: if you use Groovy ''' triple quotes then all the " quotes inside do NOT need quoting

I tened to use ''' quotes for all things graphql

File: src/test/groovy/graphql/analysis/QueryTraverserTest.groovy [hunk 2]
```diff
@@ -581,6 +581,62 @@ class QueryTraverserTest extends Specification {
...
+        QueryTraverser queryTraversal = createQueryTraversal(query, schema, [:])
+        when:
+        queryTraversal."$visitFn"(visitor)
```
Comment (bbakerman) [inline src/test/groovy/graphql/analysis/QueryTraverserTest.groovy:626]: OMG groovy -

**Review by bbakerman — APPROVED** [review]

**Comment by andimarek** [comment]
this look a bit hacky: can we just abort the traversal when we encounter a variable reference?

File: src/main/java/graphql/analysis/NodeVisitorWithTypeTracking.java [hunk 3]
```diff
@@ -233,20 +233,10 @@ public TraversalControl visitObjectField(ObjectField node, TraverserContext<Node
...
-            return true;
-        }
-        return isChildOfVariableDefinition(context.getParentContext());
-    }
-
     @Override
     protected TraversalControl visitValue(Value<?> value, TraverserContext<Node> context) {
-        if (isChildOfVariableDefinition(context)) {
-            return TraversalControl.CONTINUE;
+        if (context.getParentNode() instanceof VariableDefinition) {
```
Comment (andimarek) [inline src/main/java/graphql/analysis/NodeVisitorWithTypeTracking.java:238]: We can do even better: we can directly visitVariableDefinition and abort ( I think)

**Comment by andimarek** [comment]
@temaEmelyan is that the same as https://github.com/graphql-java/graphql-java/pull/2329 ?

**Pull request merged**